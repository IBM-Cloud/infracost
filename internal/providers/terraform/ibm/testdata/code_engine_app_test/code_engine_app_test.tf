terraform {
  required_providers {
    ibm = {
      source = "IBM-Cloud/ibm"
    }
  }
}

provider "ibm" {
  region = "us-south"
}

resource "ibm_resource_group" "test_group" {
  name = "test-resource-group"
}

resource "ibm_code_engine_project" "ce_project" {
  name              = "ce_project"
  resource_group_id = ibm_resource_group.test_group.id
}

resource "ibm_code_engine_app" "ce_app" {
  project_id              = ibm_code_engine_project.ce_project.id
  name                    = "ce-app"
  image_reference         = "icr.io/codeengine/helloworld"
  scale_initial_instances = 1
  scale_memory_limit      = "2G"
  scale_cpu_limit         = "1"
}

resource "ibm_code_engine_app" "ce_app2" {
  project_id      = ibm_code_engine_project.ce_project.id
  name            = "ce-app2"
  image_reference = "icr.io/codeengine/helloworld"
}

resource "ibm_code_engine_app" "ce_app3" {
  project_id              = ibm_code_engine_project.ce_project.id
  name                    = "ce-app3"
  image_reference         = "icr.io/codeengine/helloworld"
  scale_initial_instances = 2
  scale_memory_limit      = "2G"
  scale_cpu_limit         = "1"
}

resource "ibm_code_engine_app" "ce_app4" {
  project_id              = ibm_code_engine_project.ce_project.id
  name                    = "ce-app4"
  image_reference         = "icr.io/codeengine/helloworld"
  scale_initial_instances = 3
  scale_memory_limit      = "2000M"
  scale_cpu_limit         = "0.5"
}