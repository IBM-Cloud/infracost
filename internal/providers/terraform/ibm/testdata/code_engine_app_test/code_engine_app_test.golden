
 Name                                              Monthly Qty  Unit  Monthly Cost 
                                                                                   
 ibm_code_engine_app.ce_app                                                        
 ├─ Virtual Processor Cores (2 initial instances)         0.02  vCPU         $0.25 
 └─ RAM (2 initial instances)                             0.08  GB           $0.10 
                                                                                   
 ibm_code_engine_app.ce_app2                                                       
 ├─ Virtual Processor Cores (1 initial instances)         0.01  vCPU         $0.12 
 └─ RAM (1 initial instances)                             0.04  GB           $0.05 
                                                                                   
 ibm_code_engine_app.ce_app3                                                       
 ├─ Virtual Processor Cores (1 initial instances)         0.01  vCPU         $0.12 
 └─ RAM (1 initial instances)                             0.02  GB           $0.03 
                                                                                   
 ibm_code_engine_app.ce_app4                                                       
 ├─ Virtual Processor Cores (2 initial instances)         0.02  vCPU         $0.25 
 └─ RAM (2 initial instances)                             0.04  GB           $0.05 
                                                                                   
 ibm_code_engine_app.ce_app5                                                       
 ├─ Virtual Processor Cores (1 initial instances)         0.01  vCPU         $0.12 
 └─ RAM (1 initial instances)                           0.0195  GB           $0.03 
                                                                                   
 OVERALL TOTAL                                                               $1.12 
──────────────────────────────────
7 cloud resources were detected:
∙ 5 were estimated
∙ 2 were free:
  ∙ 1 x ibm_code_engine_project
  ∙ 1 x ibm_resource_group